Ginger/MSAccessDB/MSAccessDBCon.cs

```diff
@@ -0,0 +1,352 @@
+                    {
+                        string ColName = (string)row[0];
+                        rc.Add(ColName);
+                    }
+                }
+                catch (Exception e)
+                {
+                    Reporter.ToLog(eLogLevel.ERROR, "", e);
+                    throw (e);
+                }
+                finally
+                {
+                    reader.Close();
+                }
+            return rc;
+        }
+
+        public bool MakeSureConnectionIsOpen()
+        {
+            Boolean isCoonected = true;
+
+            if ((conn == null) || (conn.State != ConnectionState.Open))
+            {
+                isCoonected = OpenConnection(KeyvalParamatersList);
+            }
+            //make sure that the connection was not refused by the server               
+            TimeSpan timeDiff = DateTime.Now - LastConnectionUsedTime;
+            if (timeDiff.TotalMinutes > 5)
+            {
+                isCoonected = OpenConnection(KeyvalParamatersList);
+            }
+            else
+            {
+                LastConnectionUsedTime = DateTime.Now;
+            }
+            return isCoonected;
+        }
+
+        public bool OpenConnection(Dictionary<string, string> parameters)
+        {
+            
+            KeyvalParamatersList = parameters;
+            string connectConnectionString = GetConnectionString(parameters);
+            try
+            {
+                conn = new OleDbConnection(connectConnectionString);
+                conn.Open();
+                
+                if ((conn != null) && (conn.State == ConnectionState.Open))
+                {
+                    LastConnectionUsedTime = DateTime.Now;
+                    return true;
+                }
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "DB connection failed, Connection String =" + connectConnectionString, ex);
+                throw (ex);
+            }
+            return false;
+        }
+
+        public string RunUpdateCommand(string updateCmd, bool commit = true)
+        {
+            string result = "";
+            //if (conn == null) Connect();
+            if (MakeSureConnectionIsOpen())
+            {
+                using (DbCommand command = conn.CreateCommand())
+                {
+                    try
+                    {
+                        if (commit)
+                        {
+                            tran = conn.BeginTransaction();
+                            // to Command object for a pending local transaction
+                            command.Connection = conn;
+                            command.Transaction = tran;
+                        }
+                        command.CommandText = updateCmd;
+                        command.CommandType = CommandType.Text;
+
+                        result = command.ExecuteNonQuery().ToString();
+                        if (commit)
```

Comment:
if no commit done then what is the use case?

Reply:
there is a condition in a database action : if user wants to commit he can commit or he can just update